ITGlueAPI/Internal/ModuleSettings.ps1

```diff
@@ -1,56 +1,281 @@
-# Locations of settings path and file
-$ITGlueAPIConfPath = "$($env:USERPROFILE)\ITGlueAPI"
-$ITGlueAPIConfFile = "config.psd1"
-
 function Export-ITGlueModuleSettings {
+<#
+    .SYNOPSIS
+        Exports the ITGlue BaseURI, API, & JSON configuration information to file.
+
+    .DESCRIPTION
+        The Export-ITGlueModuleSettings cmdlet exports the ITGlue BaseURI, API, &
+        JSON configuration information to file.
+
+        Making use of PowerShell's System.Security.SecureString type, exporting module settings
+        encrypts your API key in a format that can only be unencrypted with the Windows account
+        that encrypted the API key.
+
+        This means that you cannot copy your configuration file to another computer or
+        use another user account and expect it to work.
+
+    .PARAMETER ITGlueAPIConfPath
+        Define the location to store the ITGlue configuration file.
+
+        By default the configuration file is stored in the following location:
+            $env:USERPROFILE\ITGlueAPI
+
+    .PARAMETER ITGlueAPIConfFile
+        Define the name of the ITGlue configuration file.
+
+        By default the configuration file is named:
+            config.psd1
+
+    .PARAMETER OpenConfigPath
+        Opens the config path location, helpful for testing
+        and validation.
+
+    .EXAMPLE
+        Export-ITGlueModuleSettings
+
+        Validates that the BaseURI, API, and JSON depth are set then exports their values
+        to the current user's ITGlue configuration file located at:
+            $env:USERPROFILE\ITGlueAPI\config.psd1
+
+    .EXAMPLE
+        Export-ITGlueModuleSettings -ITGlueAPIConfPath C:\ITGlueAPI -ITGlueAPIConfFile MyConfig.psd1 -OpenConfigPath
+
+        Validates that the BaseURI, API, and JSON depth are set then exports their values
+        to the current user's ITGlue configuration file located at:
+            C:\ITGlueAPI\MyConfig.psd1
+
+        If the configuration file is successfully created then its save location is opened up.
+
+    .NOTES
+        N\A
+
+    .LINK
+        https://github.com/itglue/powershellwrapper
+
+#>
+
+    [CmdletBinding()]
+    Param (
+        [Parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [string]$ITGlueAPIConfPath = "$($env:USERPROFILE)\ITGlueAPI",
+
+        [Parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [string]$ITGlueAPIConfFile = 'config.psd1',
+
+        [Parameter(Mandatory = $false)]
+        [switch]$OpenConfigPath
```

Comment:
The rest of the module tends to use `snake_case` instead of `PascalCase` - I recognize that it goes against Microsoft's best practices, but I think consistency within the module should take precedence. 

Adding in parameter aliases so that new people can use the `PascalCase` scheme while existing scripts fall back to `snake_case` is a good topic for a future PR, though - I'll file an Issue for it.

Reply:
I can get that switched over for consistency. How does the following look?

- itglue_api_conf_path
- itglue_api_conf_file
- open_conf_Path    (renamed config to conf)

To be honest I didn't even think about it with this section as I copied what was there previously and moved them into parameters.

```powershell
# Locations of settings path and file
$ITGlueAPIConfPath = "$($env:USERPROFILE)\ITGlueAPI"
$ITGlueAPIConfFile = "config.psd1"
```

Completely agree with

> but I think consistency within the module should take precedence.

Reply:
Those look good! Figures it'd be a holdover from some code I wrote years ago 😆